src/TicketMachine.java

```diff
@@ -75,18 +104,102 @@ Ticket chooseFilm(){ //Ticket is the return type here. It is a Class Return and
         int choice = userInput.nextInt();
         switch (choice){
             case 1: //SW4, only during testing
```

Comment:
Hard-coding the `case`s does not allow for extending the `ticketsList` with more films.
Ideally, you would do something like `return availableTicketsList.get(choice - 1)`. But maybe you already had that in mind 🙂

Reply:
Yeah, I knew I had to change this at some point, and at first I thought: well I'll just rewrite the function when more films are added. But thanks to your comment I realised that is not a good way of going about it. 
If I am  correct one wants to only have to update the ticketlist and then have all the logic work. Yes? How do I say this? Ehm, like keep things simple and one always wants to change as little as possible and always in as few places as possible and always as logically as possible, yes?

Your solution has me realising that I need to use such solutions more and more. I just got the principle of the if/while (boolean argument), but that same logic applies to the return as well. My mind still loves to go to the if/else, loops, swtich, etc to do things, but this is way more elegant :)

Thanks

Reply:
> If I am correct one wants to only have to update the ticketlist and then have all the logic work. Yes? How do I say this? Ehm, like keep things simple and one always wants to change as little as possible and always in as few places as possible and always as logically as possible, yes?

Indeed! One way of saying this is: making something scalable. 
Basically, everything is scalable as long as you throw enough money/time/effort at the problem. But the idea is that something is easily scalable, without a lot of effort. Indeed, by just adding a new film to the ticket list.

Reply:
> Your solution has me realising that I need to use such solutions more and more. I just got the principle of the if/while (boolean argument), but that same logic applies to the return as well. My mind still loves to go to the if/else, loops, swtich, etc to do things, but this is way more elegant :)

This will come naturally, the more you do it 🙂. No rush!

Reply:
> This will come naturally, the more you do it 🙂. No rush!

I know... Thanks though for the kind words :)